data/World/Deku Tree.json

```diff
@@ -54,6 +56,17 @@
     {
         "region_name": "Deku Tree Basement Ledge",
         "dungeon": "Deku Tree",
+        "locations": {
+            "Deku Tree Basement Recovery Heart 1" : "
+                deku_tree_shortcuts
+                or here(has_fire_source_with_torch or (logic_deku_b1_webs_with_bow and can_use(Bow)))",
+            "Deku Tree Basement Recovery Heart 2" : "
+                deku_tree_shortcuts
+                or here(has_fire_source_with_torch or (logic_deku_b1_webs_with_bow and can_use(Bow)))",
+            "Deku Tree Basement Recovery Heart 3" : "
+                deku_tree_shortcuts
+                or here(has_fire_source_with_torch or (logic_deku_b1_webs_with_bow and can_use(Bow)))"
+        },
```

Comment:
This works as-is. But I suggest a new region for the 2nd basement and put the 3 hearts there along with the shield/shortcuts check for continuing to Gohma.

(I don't think the here() is doing anything on these recovery hearts since whichever age burns the web is going to be able to pick up the hearts. But that's moot if you add the region.)